opentok/opentok-web-samples
# OPENTOK-45821: Add sample screen-sharing applications for Electron 17 and Electron 12-16

**What is this PR doing?**
When Electron 17 releases, a [planned change to improve application security](https://www.electronjs.org/docs/latest/breaking-changes#removed-desktopcapturergetsources-in-the-renderer) will affect users' ability to screen-share in electron.

This PR is creating a sample application for developers to reference when creating a new Electron 17 application. Since Electron 17 is still unstable and hasn't been released, a sample application for Electron versions 12-16 will also be provided.

**How should this be manually tested?**
1. clone this repo
2. open the `opentok-web-samples/Electron-Basic-Screen-Sharing/V17/README.md` file
3. follow the directions in the V17 README to launch the sample application, and confirm screen-sharing works
4. repeat the first three steps for the following repos. These repos are linked at the bottom of the README.md file.
   1. https://github.com/cpettet/opentok-web-samples/tree/Electron-SS-V01-to-V04
   2. https://github.com/cpettet/opentok-web-samples/tree/Electron-SS-V05-to-V11
   3. https://github.com/cpettet/opentok-web-samples/tree/Electron-SS-V12-to-V16
5. check for comment readability in the following files:
  - `opentok-web-samples/Electron-Basic-Screen-Sharing/V17/preload.js`, lines 6-8 and 13-14
  - `opentok-web-samples/Electron-Basic-Screen-Sharing/V17/app.js`, lines 3-5

**NOTE: some of the README links were created based on the expected URLs for the newly created files and may show a 404 when accessed**

**What are the relevant tickets?**
Resolves [OPENTOK-45821](https://jira.vonage.com/browse/OPENTOK-45821)
